### PR TITLE
chore(repo): Update Turbo & skip unaffected integration tests

### PR DESCRIPTION
**Pull request by tmilewski**
## Description

This PR will greatly reduce time waiting for GitHub Actions and the overall spend.

It fixes persistent cache misses and short-circuits E2E test runs (if not necessary) before `npm ci` and `Verdaccio`.

- Upgrades Turborepo from `^1.10` to `^2.0.3`
- Fixes expired `TURBO_TOKEN` (per GitHub settings)
- Integrates `turbo-ignore` to short-circuit E2E CI runs.
- Adds `E2E_PROJECT` to ensure that E2E tests are also cached/ignored based on the project flag.

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [x] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -137,6 +141,9 @@ jobs:
     runs-on: ${{ vars.RUNNER_MEDIUM || 'ubuntu-latest-m' }}
     timeout-minutes: ${{ vars.TIMEOUT_MINUTES_LONG && fromJSON(vars.TIMEOUT_MINUTES_LONG) || 15 }}
 
+    env:
```
Comment (tmilewski) [inline .github/workflows/ci.yml:144]: Used to explicitly set NODE_VERSION between runs. Else most runs are using `18.20.2`, while `Unit Tests (18)` is using `^18`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -156,36 +163,56 @@ jobs:
           fetch-depth: 0
           show-progress: false
 
+      - name: Task Status
+        id: task-status
+        env:
```
Comment (tmilewski) [inline .github/workflows/ci.yml:168]: Ensure `task-status` gets the same environment variables as `integration-tests`

File: turbo.json [hunk 3]
```diff
@@ -23,16 +24,10 @@
     "VERCEL",
     "VITE_CLERK_*",
     "EXPO_PUBLIC_CLERK_*",
-    "REACT_APP_CLERK_*",
-    "CI_REPO_OWNER",
-    "CI_REPO_NAME",
-    "CI_COMMIT_SHA",
```
Comment (tmilewski) [inline turbo.json:29]: `CI_COMMIT_SHA`, `CI_BRANCH, etc. being in globalDependencies will force all tasks to be invalidated on each new PR.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -69,10 +71,10 @@ jobs:
         run: npx turbo lint:publint $TURBO_ARGS --only
 
       - name: Lint types using attw
-        run: npx turbo lint:attw $TURBO_ARGS --filter=!nextjs --filter=!tanstack-start --filter=!elements --filter=!backend --only
+        run: npx turbo lint:attw $TURBO_ARGS --filter=!@clerk/nextjs --filter=!@clerk/tanstack-start --filter=!@clerk/elements --filter=!@clerk/backend --only
```
Comment (tmilewski) [inline .github/workflows/ci.yml:74]: Turbo v2 no longer infers scope.

File: .changeset/angry-pigs-beam.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+'@clerk/clerk-sdk-node': patch
+'@clerk/express': patch
+'@clerk/ui': patch
+---
+
+The following are all internal changes and not relevant to any end-user:
+
+- Upgrade Turborepo from v1 to v2.
+- Integrate `turbo-ignore` to short-circuit E2E tests
+- Add `E2E_PROJECT` to ensure that E2E tests are also cached/ignored based on the project flag.
```
Comment (nikosdouvlis) [inline .changeset/angry-pigs-beam.md:14]: We can replace this with an empty changeset as we probably don't want to release any packages as a result of this PR :) 

You can also run `npm run changeset:empty`

```suggestion
---
---
```

File: turbo.json [hunk 6]
```diff
@@ -124,59 +116,65 @@
...
     "bundlewatch": {
       "dependsOn": ["build"],
+      "env": [
+        "BUNDLEWATCH_GITHUB_TOKEN",
+        "CI_BRANCH",
+        "CI_BRANCH_BASE",
+        "CI_REPO_NAME",
+        "CI_REPO_OWNER",
+        "CI_COMMIT_SHA"
+      ],
```
Comment (nikosdouvlis) [inline turbo.json:131]: ❓ Does this whitelist these env vars for the task?

**Review by nikosdouvlis — APPROVED** [review]

**Comment by BRKalow** [comment]
@tmilewski @nikosdouvlis I'm thinking we want `@clerk/clerk-js` to be a dependency of all integration test tasks, as I believe we use the locally built version in tests.

File: turbo.json (hunk 6 above)
Comment (tmilewski) [reply turbo.json:131]: Yup

**Comment by tmilewski** [comment]
> @tmilewski @nikosdouvlis I'm thinking we want `@clerk/clerk-js` to be a dependency of all integration test tasks, as I believe we use the locally built version in tests.

Added 👍

File: packages/elements/package.json [hunk 7]
```diff
@@ -75,7 +75,7 @@
     "@radix-ui/react-slot": "^1.0.2",
     "@xstate/react": "^4.1.1",
     "client-only": "^0.0.1",
-    "xstate": "^5.13.0"
+    "xstate": "^5.13.2"
```
Comment (dimkl) [inline packages/elements/package.json:78]: ❓ why did we upgrade this dependency? This does not seem related to the PR.

**Review by dimkl — APPROVED** [review]

File: packages/elements/package.json (hunk 7 above)
Comment (tmilewski) [reply packages/elements/package.json:78]: Correct, there were a number of things that became super weird after merging the rebasing main with the Next 14 version update.
